llvm/lib/Transforms/Scalar/LoopPassManager.cpp

```diff
@@ -273,8 +273,6 @@ PreservedAnalyses FunctionToLoopPassAdaptor::run(Function &F,
            llvm::any_cast<const LoopNest *>(&IR));
     const Loop **LPtr = llvm::any_cast<const Loop *>(&IR);
     const Loop *L = LPtr ? *LPtr : nullptr;
-    if (!L)
-      L = &llvm::any_cast<const LoopNest *>(IR)->getOutermostLoop();
```

Comment:
There's an `any_cast<const LoopNest *>` in an assertion left above.

Though generally I don't get why you're removing the LoopNest support here.

Reply:
I guess i missed the line above, but unless i was misunderstood layers of template template instantiation i found no code creating Any with LoopNest when i checked before.

Reply:
To add some more context it looks like [LoopPassManager::runSinglePass](https://github.com/llvm/llvm-project/blob/d2a96d170a4faa0a6c42fe5f23c073891d6118b8/llvm/include/llvm/Transforms/Scalar/LoopPassManager.h#L364) would be the path a LoopNest Any could be created but it decayed to a Loop from a overload of getLoopFromIR that takes a LoopNest and returns a Loop.

Reply:
I see, so the code intentionally passes the outermost loop instead of the LoopNest to pass instrumentation. In that case these changes are fine.